### PR TITLE
(fix) O3-3546: Return the specific lab order being modified and not the entire list

### PR DESCRIPTION
**Pull request by arodidev**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR introduces adds functionality that filters out test types from the lab order basket when modfying an order and only lists the test type relevant to the order being modified.

## Screenshots

https://github.com/user-attachments/assets/09c4ce25-a197-4785-982f-9cc99714febb

## Related Issue

https://openmrs.atlassian.net/browse/O3-3546

### CODE REVIEW

File: packages/esm-patient-labs-app/src/lab-orders/add-lab-order/test-type-search.component.tsx [hunk 1]
```diff
@@ -67,6 +69,7 @@ function TestTypeSearchResults({ searchTerm, openOrderForm, focusAndClearSearchI
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
   const { testTypes, isLoading, error } = useTestTypes();
+  const { orders, setOrders } = useOrderBasket<LabOrderBasketItem>('labs', prepLabOrderPostData);
```
Comment (arodidev) [inline packages/esm-patient-labs-app/src/lab-orders/add-lab-order/test-type-search.component.tsx:72]: Moved the `useOrderBasket` from `TestTypeSearchResults` to its parent. It didn't seem right to have it run for every iteration of the `filteredTestTypes` considering it returns the same value.

**Comment by ibacher** [comment]
My preferred solution for this problem is [this](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1928/), i.e., just take the user directly to the modify form.

**Comment by brandones** [comment]
@ibacher We merged https://github.com/openmrs/openmrs-esm-patient-chart/pull/1928/ , does that mean we should close this out (and the ticket)?

**Comment by denniskigen** [comment]
@arodidev should we close this per @ibacher's comment above?

**Comment by arodidev** [comment]
Yes @denniskigen feel free to close this PR, thanks.

**Pull request closed without merging**